plugins/cache-ehcache/src/main/java/org/opensearch/cache/store/disk/EhcacheDiskCache.java

```diff
@@ -236,11 +253,13 @@ public V get(K key) {
         if (key == null) {
             throw new IllegalArgumentException("Key passed to ehcache disk cache was null.");
         }
-        V value;
-        try {
-            value = cache.get(key);
-        } catch (CacheLoadingException ex) {
-            throw new OpenSearchException("Exception occurred while trying to fetch item from ehcache disk cache");
+        V value = null;
+        if (keystore.contains(key.hashCode()) || keystore.isFull()) {
+            try {
+                value = cache.get(key);
```

Comment:
Do we have any numbers for `get` calls from this Ehcache based implementation ? I wanted to understand how much is the overhead of these `get` calls when key is absent in Ehcache, to see if we really need RBM here. Would expect probably some of these optimizations to be in Ehcache impl itself (but not necessary)